### PR TITLE
VACMS-12290 Remove spec tool tests

### LINKED ISSUE #8586: Update spec tool behat tests to combine content type fields tests
## Description
As a developer, I want to run ALL the fields tests at once for content types. Currently there are 12 separate tests for the following 'groups' of content:

1. Benefits Hubs - page, landing_page, basic_landing_page, support_service
2. Campaign landing page - campaign_landing_page
3. VA common - centralized_content
4. LC (Resources and Support) - checklist, faq_multiple_q_a, media_list_images, media_list_videos, q_a, support_resources_detail_page, step_by_step
5. User guides (KB) - documentation_page
6. VAMC - event, event_listing, health_services_listing, leadership_listing, press_release, press_releases_listing, person_profile, story_listing, news_story, health_care_region_detail_page, health_care_local_facility, health_care_local_health_service, vha_facility_nonclinical_service, health_care_region_page, full_width_banner_alert, vamc_system_billing_insurance, regional_health_care_service_des, locations_listing, vamc_system_medical_records_offi, vamc_operating_status_and_alerts, vamc_system_policies_page, vamc_system_register_for_care
7. Alerts - banner, promo_banner
8. NCA facility - nca_facility
9. Outreach hub - office, outreach_asset, publication_listing
10. VA forms - va_form
11. VBA facility - vba_facility
12. Vet Center - vet_center, vet_center_cap, vet_center_facility_health_servi, vet_center_locations_list, vet_center_mobile_vet_center, vet_center_outstation

The tests within the new spec tool in airtable already account for this combined test using the output from the new [spec tool](https://airtable.com/appSg80z9nVWoesxn/tbltV84WALyKvpXuc/viwqLFrgfQleFk9Zp?blocks=hide) (though we'll need to add the intro text and list of content types in, if needed).

## Acceptance Criteria
- [ ] New test for all content type fields is created
- [ ] Test runs successfully using output generated in new spec tool

### CMS Team
Please check the team(s) that will do this work. 

- [ ] `CMS Program`
- [x] `Platform CMS Team`
- [ ] `Sitewide CMS Team ` (leave Sitewide unchecked and check the specific team instead)
  - [x] `⭐️ Content ops`
  - [ ] `⭐️ CMS experience`
  - [ ] `⭐️ Offices`
  - [ ] `⭐️ Product support`
  - [ ] `⭐️ User support`

### LINKED ISSUE #9497: Automate syncing of CMS Content Model to Airtable to replace Content Spec behat test
## Background
We discussed in a Spike here https://github.com/department-of-veterans-affairs/va.gov-cms/issues/8665.

### User Story or Problem Statement
As a CMS Development Team Member I need to see the current state of our deployed content model in the "VA.gov CMS content model documentation" Airtable Base so that I can create additional views, add metadata and collaborate with others on content model changes.

### Affected users and stakeholders

* Product Owners & Managers
* Software Engineers
* UX Engineers
* Delivery Managers

### Hypothesis
We believe that removing the Behat Spec Tool tests and instead, syncing the deployed CMS config directly to the "VA.gov CMS content model documentation" Airtable Base will greatly improve the developer workflow, improve test suite run times, and maintain clarity into the currently deployed content model. 

We will know this is true when we:
* Measure test times before and after
* Developers stop having merge conflicts with the Behat Spec Tool tests
* The deployed Content Model state is accurately reflected in the "VA.gov CMS content model documentation" Airtable Base on an hourly basis

## Assumptions
* That we have all of the state needed from the CMS *.yml config files (PoC proves this already)
* This will improve Developer Experience
* This will decrease test suite run times

## Acceptance Criteria
- [ ] #9506
- [ ] The desired CMS config state is synced:
  - [ ] #9505
  - [ ] Bundle Types
  - [ ] Bundles
  - [ ] Field Types
  - [ ] Fields
  - [ ] Views/Views displays
  - [ ] Modules
  - [ ] Image styles, Image effects
  - [ ] Menus
  - [ ] Form widgets
  - [ ] Workflows, Workflow states, Workflow transitions
- [ ] Measure speed of tests suite runs to see if overall time is decreased
- [ ] Ask developers if their workflow is improved

## Implementation notes

* Current plan is to use a GitHub Action and take the CMS config/*.yml (convert to CSV or JSON) and send it to Airtable with their [API](https://www.airtable.com/developers/scripting/api).
  * Do this on an hourly basis
* The "User roles" table might be the easiest to get started with because it doesn't have any relationships. 
  * It is okay to change the ID on this column for any of the tables, e.g. make it the same as the machine name. Currently the ID field on all tables is the Name or Label field. 
  * We have a TEST Base here https://airtable.com/appZ4csS08RldVOIj/tbl8tQ719TAitQyrz/viwDZf3ewK3bRVB97 that we must test on, ask @msbtterswrth for a "Creator Invite to the Test Content Model Base".
  * Ask @msbtterswrth or @kevwalsh for clarifying questions that were missed here
 
## Design principles

Veteran-centered
- [x] `Single source of truth`: Increase reliability and consistency of content on VA.gov by providing a single source of truth.
- [ ] `Accessible, plain language`: Provide guardrails and guidelines to ensure content quality.
- [x] `Purposely structured content`: Ensure Content API can deliver content whose meaning matches its structure.
- [x] `Content lifecycle governance`: Produce tools, processes and policies to maintain content quality throughout its lifecycle.

Editor-centered
- [ ] `Purpose-driven`: Create an opportunity to involve the editor community in VA’s mission and content strategy goals.
- [ ] `Efficient`: Remove distractions and create clear, straightforward paths to get the job done.
- [ ] `Approachable`: Offer friendly guidance over authoritative instruction.
- [ ] `Consistent`: Reduce user’s mental load by allowing them to fall back on pattern recognition to complete tasks.
- [ ] `Empowering`: Provide clear information to help editors make decisions about their work.

### CMS Team
Please check the team(s) that will do this work.

- [ ] `Program`
- [x] `Platform CMS Team`
- [ ] `Sitewide Crew`
- [x] `⭐️ Sitewide CMS`
- [ ] `⭐️ Public Websites`
- [ ] `⭐️ Facilities`
- [ ] `⭐️ User support`

### LINKED ISSUE #11938: Ensure formatting for Spec tool files in pre-commit check and GitHub Action.
## Description
These can cause a lot of PR aggravation.

Need to look at current format and figure out how to efficiently detect departures.

## Acceptance Criteria
- [ ] Testable_Outcome_X
- [ ] Testable_Outcome_Y
- [ ] Testable_Outcome_Z
- [ ] Requires design review

### LINKED ISSUE #12290: Remove spec tool behat tests
Blocked by MVP

Once the MVP for Content Model Documentation is complete  and the custom content currently residing in the air table has been added to the documentation entity the spec tests and the spectool airtable can be decomissioned.

closes #8586
closes #9497 
closes #11938

### PR DESCRIPTION
**Pull request by swirtSJW**
## Description

closes #12290

Removes the need for the following 
- closes #8586
- closes #9497
- closes #11938

## QA steps
- [ ] automated tests pass (since this is just test removal)
- [x] documentation is removed (there are no references to spectool markdown files.

### Definition of Done

- [x] Documentation has been updated, if applicable.
- [ ] Need communication after merge.

### Select Team for PR review

- [ ] `CMS Team`
- [ ] `Public websites`
- [ ] `Facilities`
- [ ] `User support`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing annoucement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by swirtSJW** [comment]
It looks like some of our reliance on Behat was  using the dependencies of the spec tool.  :(

```
+ behat
./tests/scripts/behat-tests.sh: line 13: behat: command not found
Script ! ./scripts/should-run-directly.sh || ./tests/scripts/behat-tests.sh handling the va:test:behat event returned with error code 127
```

and

```
Error in bootstrap script: Error:
Class "Behat\Mink\Element\TraversableElement" not found
```

**Comment by swirtSJW** [comment]
False failure on the cypress side
image

**Comment by swirtSJW** [comment]
Here is the behat failure.  Pretty non-descript
```
Script ! ./scripts/should-run-directly.sh || ./tests/scripts/behat-tests.sh handling the va:test:behat event returned with error code 1
```

**Comment by swirtSJW** [comment]
Running locally I get 
image

**Comment by swirtSJW** [comment]
I am seeing a ton of this failure. 
```
  ╳  Type error: Drupal\flag\FlagCountManager::resetLoadedCounts(): Argument #1 ($entity) must be of type Drupal\Core\Entity\EntityInterface, null given, called in /var/lib/tugboat/docroot/modules/contrib/flag/src/FlagCountManager.php on line 246 (Behat\Testwork\Call\Exception\FatalThrowableError)
```

It is either from one of the individual bumps, or I am missing one of the connector modules..  Will hunt for it in the morining

File: tests/behat/behat.yml [hunk 1]
```diff
@@ -4,15 +4,9 @@ default:
...
       paths:
-        - "%paths.base%/drupal-spec-tool"
         - "%paths.base%/features-default"
         - "%paths.base%/features"
       contexts:
-        - Acquia\DrupalSpecTool\Context\AccessControlContext
-        - Acquia\DrupalSpecTool\Context\MediaContext
-        - Acquia\DrupalSpecTool\Context\MenuContext
-        - Acquia\DrupalSpecTool\Context\ViewsContext
-        - Acquia\DrupalSpecTool\Context\WorkflowContext
```
Comment (swirtSJW) [inline tests/behat/behat.yml:15]: I think one of these contexts is actually enabling one of our non-spec tool tests to run.

Comment (swirtSJW) [reply tests/behat/behat.yml:15]: Confirmed  https://github.com/department-of-veterans-affairs/va.gov-cms/issues/12290#issuecomment-1448662253

**Review by timcosgrove — DISMISSED** [review]
Approved on code review and on tests passing on Tugboat.

**Comment by swirtSJW** [comment]
image

fixing conflict

**Pull request merged**